remote/src/main/java/com/playone/mobile/remote/model/PlayoneModel.kt

```diff
@@ -1,84 +1,42 @@
-//                            "name" to name,
-//                            "description" to description,
-//                            "host" to host,
-//                            "date" to date,
-//                            "updated" to updated,
-//                            "address" to address,
-//                            "latitude" to latitude,
-//                            "longitude" to longitude,
-//                            "limit" to limit,
-//                            "level" to level,
-//                            "userId" to userId)
-//
-//    fun copy(
-//        id: String? = null,
-//        name: String? = null,
-//        description: String? = null,
-//        date: Long? = null,
-//        updated: Long? = null,
-//        address: String? = null,
-//        longitude: Double? = null,
-//        latitude: Double? = null,
-//        limit: Int? = null,
-//        level: Int? = null,
-//        host: String? = null,
-//        userId: String? = null
-//    ) = PlayoneModel().also {
-//        it.id = id ?: this.id
-//        it.name = name ?: this.name
-//        it.description = description ?: this.description
-//        it.date = date ?: this.date
-//        it.updated = updated ?: this.updated
-//        it.address = address ?: this.address
-//        it.longitude = longitude ?: this.longitude
-//        it.latitude = latitude ?: this.latitude
-//        it.limit = limit ?: this.limit
-//        it.level = level ?: this.level
-//        it.host = host ?: this.host
-//        it.userId = userId ?: this.userId
-//    }
-//}
-
 data class PlayoneModel(
-    var id: String = "",
-    var name: String = "",
-    var description: String = "",
-    var date: Long = -1L,
-    var updated: Long = -1L,
-    var address: String = "",
-    var longitude: Double = -1.0,
-    var latitude: Double = -1.0,
-    var limit: Int = -1,
-    var level: Int = -1,
-    var host: String = "",
-    var userId: String = ""
+    var id: String = defaultStr,
+    var name: String = defaultStr,
+    var description: String = defaultStr,
+    var date: Long = defaultLong,
+    var updated: Long = defaultLong,
+    var address: String = defaultStr,
+    var longitude: Double = defaultDouble,
+    var latitude: Double = defaultDouble,
+    var limit: Int = defaultInt,
+    var level: Int = defaultInt,
+    var host: String = defaultStr,
+    var userId: String = defaultStr
 ) {
 
+    var _id by NotValidVar(id)
+    var _name by NotValidVar(name)
+    var _description by NotValidVar(description)
+    var _date by NotValidVar(date)
+    var _updated by NotValidVar(updated)
+    var _address by NotValidVar(address)
+    var _longitude by NotValidVar(longitude)
+    var _latitude by NotValidVar(latitude)
+    var _limit by NotValidVar(limit)
+    var _level by NotValidVar(level)
+    var _host by NotValidVar(host)
+    var _userId by NotValidVar(userId)
+
     fun toMap() = hashMapOf("name" to name,
```

Comment:
this is function is necessary?

Reply:
Yes, we need this function for updating.